store/searchtest/user_layer.go

```diff
@@ -238,15 +238,15 @@ func testHonorChannelRestrictionsAutocompletingUsers(t *testing.T, th *SearchTes
 	})
 	t.Run("Autocomplete users with all channels restricted", func(t *testing.T) {
 		options := createDefaultOptions(true, false, false)
-		options.ViewRestrictions = &model.ViewUsersRestrictions{Channels: []string{}}
+		options.ViewRestrictions = &model.ViewUsersRestrictions{Teams: []string{}, Channels: []string{}}
```

Comment:
You have to activate these tests for all the search engines doing `[]string{ENGINE_ALL}`

[here](https://github.com/mattermost/mattermost-server/pull/15912/files#diff-2c1c4b1eaee0b08c3dc2a4eeb4a0e01a8ff63024b1f701e82ea1519df4994559R23) and [here](https://github.com/mattermost/mattermost-server/pull/15912/files#diff-2c1c4b1eaee0b08c3dc2a4eeb4a0e01a8ff63024b1f701e82ea1519df4994559R28)

Reply:
The links are not working, but I assume that you are referring to the updated changes 😃